projects/vdk-plugins/vdk-meta-jobs/tests/test_meta_dag.py

```diff
@@ -67,3 +70,42 @@ def mock_get_finished_job_names(*args, **kwargs):
 
     # check for busyloop (this would have been called hundreds of times if there is busyloop bug)
     assert calls[0] <= 4
+
+
+def test_build_dag_over_max_starting_jobs_sad_case():
+    jobs = [
+        {"job_name": f"job{i}", "depends_on": [] if i == 1 else [f"job1"]}
+        for i in range(1, 18)
+    ]
+
+    dag = MetaJobsDag("team")
+    dag.build_dag(jobs)
+
+    dag._job_executor = MagicMock(spec=TrackingDataJobExecutor)
+    dag._job_executor.get_finished_job_names.side_effect = [
+        ["job1"],
+        [
+            "job2",
+            "job3",
+            "job4",
+            "job5",
+            "job6",
+            "job7",
+            "job8",
+            "job9",
+            "job10",
+            "job11",
+            "job12",
+            "job13",
+            "job14",
+            "job15",
+            "job16",
+            "job17",
+        ],
+    ]
+
+    dag.execute_dag()
+
+    assert [
+        call(job["job_name"]) for job in jobs[:17]
+    ] == dag._job_executor.start_job.call_args_list
```

Comment:
I am not sure how do we verify that at max VDK_META_JOBS_MAX_CONCURRENT_STARTING_JOBS number of jobs are started and the last few are pending. 

Also to simplify the test (less jobs) and also not break the test if the default is changed, It's good idea to explicitly set the max concurrent jobs setting:  

You can do that with 
```
@mock.patch.dict(
            os.environ,  { xxx: "val" }   )
```

Reply:
I added the mock patch with the default value of 15 and still figuring out the way to verify the last few pending jobs.

Reply:
To make the test easier to read/consume. lower the value to 5. 

You can checkout https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-meta-jobs/tests/test_meta_job.py for ideas.